### PR TITLE
Adds Upgraded Cybernetic Heart

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Adds an upgraded cybernetic heart.

Very largely inspired by Goon, so credit to them for the idea.

The upgraded cybernetic heart is a heart that is capable of restarting itself, should you have a heart attack. It also gives a bit of stun reduction and increases your stamina regen.

As a downside to this, however though, it's vulnerable to two things: electric shock and EMP. 

An EMP won't just stop the heart, it'll break it; you'll have to print a new one or repair the damage with nanopaste; simply restarting it, post EMP, isn't an option. Broken cybernetic hearts will not grant stun reduction nor attempt to restart themselves from a heart attack.

The other weakness, electric shock, occurs any time you're shocked by anything, stunned by a stun baton, or defibbed. Defibs are particularly nasty to upgraded cybernetic hearts, so be careful.

Can be made at R&D when upgraded cybernetic lungs can be made.

pics: 
img

Odds and ends
- Can now repair robotic organs with nanopaste
- rejuvinating an organ now calls `surgerize` on it to ensure edge cases for organs are fixed (like IPC arm charger)

:cl: Fox McCloud
add: Adds upgraded cybernetic heart to R&D
add: can now repair robotic organs, outside of the body, with nanopaste
/:cl:

### CODE REVIEW

**Comment by Aurorablade** [comment]
When can i replace hearts with a potato fox?

File: code/modules/mob/living/carbon/carbon.dm [hunk 1]
```diff
@@ -1116,3 +1117,7 @@ so that different stomachs can handle things in different ways VB*/
 	if(I.flags_inv & HIDEMASK || forced)
 		update_inv_wear_mask()
 	update_inv_head()
+
+/mob/living/carbon/proc/shock_internal_organs(intensity)
```
Comment (Aurorablade) [inline code/modules/mob/living/carbon/carbon.dm]: I approve this proc name

File: code/game/objects/items/weapons/stunbaton.dm [hunk 2]
```diff
@@ -172,6 +172,10 @@
 			playsound(L, 'sound/weapons/genhit.ogg', 50, 1)
 			return
 
+	if(iscarbon(L))
```
Comment (Aurorablade) [inline code/game/objects/items/weapons/stunbaton.dm]: Now this shouldn't need to be anything to consider as shock_internal organs parses through the list of internal organs...BUT AS THIS IS BYOND CODE...would adding an hasorgans check here be safe?

File: code/modules/surgery/organs/heart.dm [hunk 3]
```diff
@@ -146,5 +146,104 @@
...
+	icon_base = "heart-c-u"
+	dead_icon = "heart-c-u-off"
+	var/emagged = FALSE
+	var/attempted_restart = FALSE
+
+/obj/item/organ/internal/heart/cybernetic/upgraded/on_life()
+	if(!ishuman(owner))
+		return
+	if(!(status & ORGAN_DEAD) && !attempted_restart && !beating)
+		to_chat(owner, "<span class='warning'>Your [name] detects a cardiac event and attempts to return to its normal rhythm!</span>")
```
Comment (Aurorablade) [inline code/modules/surgery/organs/heart.dm]: Now this is just me going for aesthtic..but it should be phrased as a little robot voice in your head informing you of heart errors (not all of them useful like true errors)

**Comment by Breenland** [comment]
I was a bit worried with all the upgrades and what it’d do to obsolete the CNS Rebooter, but with the downsides I feel like that’s a fair trade. Does defibbing one of these just cause extra damage to the organ?

**Comment by Fox-McCloud** [comment]
It has a high chance of causing their heart to fail.

**Comment by Landerlow** [comment]
I think this would be a nice addition.

**Pull request merged**